[Tech] Skip empty arguments in sideload and frontend command launcher

rdbrschf opened the pull request:
If the additional command line arguments are unset/set to an empty string, the wine command launcher will likewise pass an empty argument to the launched application.

While that is usually harmless, some applications try to parse this argument and (rightfully) complain and usually exit with an error.

Make sure that we only pass additional arguments if they are actually set, fixing that issue.

This has been encountered with Palia's launcher.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [X] Tested the feature and it's working on a current and clean install.
- [X] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

rdbrschf commented:
This needs a rework, as discussed with @CommandMC on Discord. Turning into a draft for now.

rdbrschf commented:
v2: modified the correct location (sideloaded launch function), modified frontend update bypass code to not add spurious whitespace characters (although the actual runners should generally handle this gracefully via `shlex.split()`).

CommandMC commented on src/backend/storeManagers/storeManagerCommon/games.ts:
```diff
@@ -160,15 +161,17 @@ export async function launchGame(
     if (isNative) {
       logInfo(
-        `launching native sideloaded game: ${executable} ${launcherArgs ?? ''}`,
+        `launching native sideloaded game: ${executable} ${extraArgs.join(
+          ' '
+        )}`,
```
Prettier sure does destroy the formatting here
Creating a separate `const extraArgsJoined = extraArgs.join(' ')` above and using it here might be "cleaner"

flavioislima commented:
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/main/doc/cla.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

rdbrschf commented:
I have read the CLA Document and I hereby sign the CLA.

rdbrschf commented:
recheck

rdbrschf commented:
Reworked this.

Fixed the syntax nit by using another variable as proposed, dropped the second commit (obsolete, newer code passes update skipping mechanic in a different way to the legendary component).
